### ISSUE TITLE
Rebuild recycling recipes when mods change

### ISSUE THREAD

**Issue by dgw** [DRY411S/Recycling-Machines#34]
This came up a couple of times in the forum thread, and I think I found a way to handle it: `LuaBootstrap::on_configuration_changed(f)`, confirmed available in both [0.14.23](http://lua-api.factorio.com/0.14.23/LuaBootstrap.html#LuaBootstrap.on_configuration_changed) and [0.15.21](http://lua-api.factorio.com/0.15.21/LuaBootstrap.html#LuaBootstrap.on_configuration_changed). (It's even available in [0.13.20](http://lua-api.factorio.com/0.13.20/LuaBootstrap.html#LuaBootstrap.on_configuration_changed), if you're still supporting that version of Factorio with this mod.)

**Comment by DRY411S** [comment]
I will take a look at this. I believe I looked at it in previous factorio versions.

If I register a function for this, I believe that when the function runs is 'too late' in the factorio bootstrap sequence. The mod can only build the data structures it requires in the 'raw data' stage of factorio, when it adds Recycling recipes.

If I understand this bootstrap correctly, then the function that I declare will run after the raw data stage, when all recipes are read-only to mods.

**Comment by DRY411S** [comment]
OK, I think i know now why this cannot be used. The event that causes the function to be called is when a player loads a previously saved map, and the map was used with a different game/mod versions, or mods have been added/removed.

At this point, I believe that the recipes are part of the map zip file and read-only. But I'll seek advice from the modding community about this.

**Comment by DRY411S** [comment]
Question asked at https://forums.factorio.com/viewtopic.php?t=50065

I'll be very embarrassed if it's as simple as it would appear to be.

**Comment by DRY411S** [comment]
Closing this issue. This MOD does add new recipes for any mods added to saved games, after the game was started.

**Comment by dgw** [comment]
Added a new mod I'd never even looked at before to my save under Factorio 0.15.23 with ZRecycling 0.15.4, and upon completing the research for a new piece of tech the recycling recipe was unlocked along with the regular recipe for that item.

This did not work when I added new mods a few Factorio patches back, so I can only guess this was a problem in the game that has since been fixed. Unlike the items covered by #35, the items in that instance that did not have recipes generated did appear in the recycling menu after tricking the game into thinking the mod was updated, as described in the forum thread.

Anyway, this does appear fixed.
